cmd/oras/internal/display/metadata/tree/discover.go

```diff
@@ -66,18 +73,118 @@ func (h *discoverHandler) OnDiscovered(referrer, subject ocispec.Descriptor) err
 	}
 	referrerNode := node.AddPath(referrer.ArtifactType, referrer.Digest)
 	if h.verbose {
-		for k, v := range referrer.Annotations {
+		addDescriptorInfo(referrer, referrerNode)
+
+		if err := addAnnotationsInfo(referrer, referrerNode); err != nil {
+			return err
+		}
+
+		if err := addLayersInfo(referrer, referrerNode, ctx, target, platform); err != nil {
+			return err
+		}
+	}
+	h.nodes[referrer.Digest] = referrerNode
+	return nil
+}
+
+func addDescriptorInfo(referrer ocispec.Descriptor, referrerNode *tree.Node) {
+	descriptorNode := referrerNode.AddPath("Descriptor")
+	descriptorNode.AddPath(strings.TrimSpace("MediaType: " + referrer.MediaType))
+	descriptorNode.AddPath(strings.TrimSpace("Size: " + strconv.FormatInt(referrer.Size, 10)))
+}
+
+func addAnnotationsInfo(referrer ocispec.Descriptor, referrerNode *tree.Node) error {
+	if len(referrer.Annotations) != 0 {
+		annotationsNode := referrerNode.AddPath("Annotations")
+		annotationsKeys := sortedKeys(referrer.Annotations)
+		for _, k := range annotationsKeys {
+			v := referrer.Annotations[k]
```

Comment:
Seems like it would be easier to unit test this stuff if it was a wrapper for referrerNode and these were addDescriptor, addAnnotations, etc.